src/Stratis.FederatedPeg.Features.FederationGateway/Wallet/FederationWalletManager.cs

```diff
@@ -992,12 +979,47 @@ private void UpdateWhenChainDownloaded(IEnumerable<FederationWallet> wallets, Da
                 TimeSpans.FiveSeconds);
         }
 
+        /// <inheritdoc />
+        public void EnableFederation(string password)
+        {
+            Guard.NotEmpty(password, nameof(password));
+
+            // Protect against de-activation if the federation is already active.
+            if (this.isFederationActive)
+            {
+                this.logger.LogTrace("(-):[FEDERATION_ALREADY_ACTIVE]");
```

Comment:
excellent. Maybe do a `this.logger.LogWarning` here.

Reply:
Done.